### PR TITLE
Forms Automated Test - LPS-131240 | master

### PR DESCRIPTION
**Pull request by Jamersonze**

### CODE REVIEW

**Comment by brittneyq** [comment]
ci:test:sf

**Comment by brittneyq** [comment]
ci:test:relevant

**Review by brittneyq — COMMENTED** [review]
@natocesarrego @diogocordeiro this should be ready for review!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase [hunk 1]
```diff
@@ -132,12 +132,64 @@ definition {
...
+
+		Click(
+			key_fieldName = "Paragraph",
+			locator1 = "FormFields#PARAGRAPH_TITLE");
+
+		SelectFrame.selectFrameNoLoading(locator1 = "CKEditor#BODY_FIELD_IFRAME");
+
+		AssertElementPresent(
+			url = "http://www.liferay.com",
+			locator1 = "FormFields#PARAGRAPH_LINK_TEXT_BODY");
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase]: @Jamersonze, I think that all these 3 steps can be handled at once by `FormFields.viewParagraphField`, can you please confirm?

Maybe, this macro just need a little change to attend your need. :wink:

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase [hunk 2]
```diff
@@ -132,12 +132,64 @@ definition {
 	}
 
 	@description = "Verify that a link to a URL can be added to the Body Text"
-	@ignore = "true"
 	@priority = "5"
 	test CanSupportALinkToAURL {
 		property portal.acceptance = "false";
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase:137]: Can you please add `property forms.functional = "true";` here following alphabetic order?

I mean, something like:
```
property forms.functional = "true";
property portal.acceptance = "false";
```

Thanks.

Comment (Jamersonze) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase:138]: Fixed

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase (hunk 1 above)
Comment (Jamersonze) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase]: Fixed

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase [hunk 3]
```diff
@@ -132,12 +132,64 @@ definition {
...
+
+		FormsAdminNavigator.gotoAddForm();
+
+		Form.editName(formName = "Form 1");
+
+		Form.gotoAddField(
+			fieldPositionNumber = "1",
+			fieldType = "Paragraph");
+
+		Form.editFieldLabelMetal(fieldValue = "Paragraph");
```
[the remark this answers is not in the document]
Comment (Jamersonze) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase]: Fixed

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase [hunk 4]
```diff
@@ -132,12 +132,64 @@ definition {
...
+
+		Click(locator1 = "CKEditor#OK_BUTTON");
+
+		Click(locator1 = "Button#SAVE");
+
+		Alert.viewSuccessMessage();
+
+		Click(locator1 = "Icon#BACK");
+
+		CommerceNavigator.gotoEntry(entryName = "Form 1");
```
[the remark this answers is not in the document]
Comment (Jamersonze) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase]: Fixed

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase [hunk 5]
```diff
@@ -132,12 +132,64 @@ definition {
...
+			category = "Content &amp; Data",
+			portlet = "Forms");
+
+		FormsAdminNavigator.gotoAddForm();
+
+		Form.editName(formName = "Form 1");
+
+		Form.gotoAddField(
+			fieldPositionNumber = "1",
+			fieldType = "Paragraph");
```
[the remark this answers is not in the document]
Comment (Jamersonze) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase]: Fixed

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormFields.path [hunk 6]
```diff
@@ -344,6 +344,26 @@
...
+<tr>
+	<td>PARAGRAPH_LINK_TEXT_INPUT</td>
+	<td>//div[@class='cke_dialog_body']//label[text()='${fieldLabel}']//..//input[@type='text']</td>
+	<td></td>
+</tr>
+<tr>
+	<td>PARAGRAPH_LINK_TEXT_BODY</td>
+	<td>//p//a[@href='${url}']</td>
+	<td></td>
+</tr>
```
[the remark this answers is not in the document]
Comment (Jamersonze) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormFields.path]: Fixed

**Comment by Jamersonze** [comment]
Fixed

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormFields.path (hunk 6 above)
[the remark this answers is not in the document]
Comment (brittneyq) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormFields.path]: Hi @Jamersonze, could you please sort it alphabetically again? PARAGRAPH_LINK_TEXT_INPUT and PARAGRAPH_LINK_TEXT_BODY should be switched to remain in alphabetical order. Thank you!

**Comment by brittneyq** [comment]
Hi @Jamersonze, please fix your file conflicts. Also please rearrange PARAGRAPH_LINK_TEXT_INPUT and PARAGRAPH_LINK_TEXT_BODY to be in alphabetical order in the path file.

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormFields.path (hunk 6 above)
[the remark this answers is not in the document]
Comment (Jamersonze) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormFields.path]: Fixed

**Comment by brittneyq** [comment]
ci:test:sf

**Comment by brittneyq** [comment]
Hi @Jamersonze, please don't delete `FormFields#TOOLBAR_ANY_ITEM_BUTTON` path in FormFields as this was already merged. Also since your test should use that path as well, please change it in your test (since it is currently referencing CKEditor). In addition, since TOOLBAR_ANY_ITEM_BUTTON is used in FormFields already, please remove your changes to CKEditor.path and remove that file from your commit as well.

**Comment by Jamersonze** [comment]
Fixed

**Comment by brittneyq** [comment]
ci:test:sf

**Comment by brittneyq** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormFields.path [hunk 7]
```diff
@@ -355,8 +355,18 @@
 	<td></td>
 </tr>
 <tr>
-	<td>TOOLBAR_ANY_ITEM_BUTTON</td>
-	<td>//a[contains(@title,"${key_titleName}")]</td>
```
Comment (emanuelfisilva) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/forms/FormFields.path:359]: Hello @Jamersonze you should not change existent PATHs due to the risk of breaking already existing tests which uses these paths. 
I've changed it back to this value and properly called it on your test.
Please see: https://github.com/emanuelfisilva/liferay-portal/pull/167/commits/80a8fc79bb17f6c461612fc84b590883d2e7b1ee

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase [hunk 8]
```diff
@@ -186,12 +186,55 @@ definition {
...
+
+		Form.gotoAddField(
+			fieldPositionNumber = "0",
+			fieldType = "Paragraph");
+
+		Click(
+			key_titleName = "Link",
+			locator1 = "FormFields#TOOLBAR_ANY_ITEM_BUTTON");
+
+		Click(
```
Comment (emanuelfisilva) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsParagraphField.testcase:210]: Hello @Jamersonze this invocation of Click macro and the next are not necessary as they're not setting the values, they're only clicking on this locators, thus I removed it.
Please see: https://github.com/emanuelfisilva/liferay-portal/pull/167/commits/ab699f7fd710b0b25ca55956ef8bab175038f10b

**Comment by emanuelfisilva** [comment]
Thanks @Jamersonze . Good work!
Please follow my changes: https://github.com/emanuelfisilva/liferay-portal/pull/167

**Pull request closed without merging**